python/mxnet/io.py

```diff
@@ -305,6 +310,14 @@ def provide_data(self):
                        for r, i in zip(self.rename_data, self.iters)], [])
 
     @property
+    def provide_data_type(self):
```

Comment:
`(name, shape, type)` will probably break existing code. Another option might be use attrs for the input variables.